msfextract

```diff
@@ -0,0 +1,240 @@
+  unzip_test = run_cmd("unzip")
+  unless unzip_test != nil
+    raise RuntimeError, "unzip command not found."
+  end
+
+  d2j_test = run_cmd("d2j-dex2jar")
+  unless d2j_test != nil
+    raise RuntimeError, "d2j-dex2jar command not found."
+  end
+
+  java_test = run_cmd("java")
+  unless java_test != nil
+    raise RuntimeError, "java command not found."
+  end
+
+  apktool_test = run_cmd("apktool")
+  unless apktool_test != nil
+    raise RuntimeError, "apktool command not found."
+  end
+end
+
+def parse_args(args)
+  opts = {}
+  opt = OptionParser.new
+  banner = "MsfExtract - a tool to extract configbytes from an android payload.\n"
+  banner << "Usage: #{$0} < -a apk-file -o path -j path > [options]\n"
+  banner << "Example: #{$0} -a metasploit.apk -o /root/configbytes.txt -j /root/fernflower.jar"
+  opt.banner = banner
+  opt.separator('')
+  opt.separator('Options:')
+
+  opt.on('-a', '--apk             <path>', String, 'Specify apk to extract configbytes') do |a|
+    opts[:apk] = a
+  end
+
+  opt.on('-o', '--out             <path>', String, 'Save configbytes to a file') do |o|
+    opts[:out] = o
+  end
+
+  opt.on('-j', '--jar             <path>', String, 'Specify fernflower path') do |j|
+    opts[:jar] = j
+  end
+
+  opt.on('-v', '--verbose', 'Displays verbose output') do
+    opts[:verbose] = true
+  end
+
+  opt.on('-k', '--keep', 'Keep working directory') do
+    opts[:keep] = true
+  end
+
+
+  opt.on_tail('-h', '--help', 'Show this message') do
+    raise HelpError, "#{opt}"
+  end
+
+  begin
+    opt.parse!(args)
+  rescue OptionParser::InvalidOption => e
+    raise UsageError, "Invalid option\n#{opt}"
+  rescue OptionParser::MissingArgument => e
+    raise UsageError, "Missing required argument for option\n#{opt}"
+  end
+
+  if opts.empty?
+    raise UsageError, "No options\n#{opt}"
+  end
+
+  if opts[:apk].nil?
+    raise UsageError, "Missing required argument apk file\n#{opt}"
+    exit(1)
+  end
+
+  if opts[:jar].nil?
+    raise UsageError, "Missing required argument fernflower path\n#{opt}"
+    exit(1)
+  end
+
+  opts
+end
+
+begin
+  options = parse_args(ARGV)
+rescue HelpError => e
+  $stderr.puts e.message
+  exit(1)
+rescue MsfExtractError => e
+  $stderr.puts "Error: #{e.message}"
+  exit(1)
+end
+
+begin
+  check_tools()
+
+  temp_dir = Dir.mktmpdir
+  temp_dir << "/"
+
+  if options[:out].nil?
+    $stderr.puts "[-] No output option selected, working in #{temp_dir}"    
+    output = "#{temp_dir}configbytes.txt"
+    options[:keep] = true
+  else
+    output = options[:out]
+  end
+
+  apk = options[:apk]
+  zip_file = "#{temp_dir}temp.zip"
+  temp_apk = "#{temp_dir}temp.apk"
+  FileUtils.cp(apk, temp_apk)
+
+  $stderr.puts "[+] Renaming apk file to zip file"
+  File.rename(temp_apk, zip_file)
+
+  $stderr.puts "[+] Using unzip on zip file for dex file"
+  unzip_cmd = run_cmd("unzip #{temp_dir}temp.zip -d #{temp_dir}")
+  if options[:verbose] then $stderr.puts "#{unzip_cmd}" else unzip_cmd end
```

Comment:
If you want to print the command that's being run the it should probably:
1) Be before the command is run so if there's an error the user know what was going on
2) Be a part of the `run_cmd` function itself so you don't need to repeat this pattern. If you only want to do it sometimes you could add a keyword argument to `run_cmd`.